go/libraries/doltcore/sqle/dolt_session.go

```diff
@@ -454,6 +460,33 @@ func (sess *DoltSession) SetRoot(ctx *sql.Context, dbName string, newRoot *doltd
 	return nil
 }
 
+func (sess *DoltSession) GetTempTableRootValue(ctx *sql.Context, dbName string) (*doltdb.RootValue, error) {
```

Comment:
This still isn't good enough. You need to only create a temp root / edit session when a temp table is created. The way this is written now, you are doing this on every table lookup, so you'll create a temp root for every session still